xrtl/port/windows/gfx/es3/wgl_platform_context.cc

```diff
@@ -0,0 +1,521 @@
+                                    reinterpret_cast<HWND>(native_window),
+                                    std::move(share_group))) {
+    LOG(ERROR) << "Unable to initialize the EGL platform context";
+    return nullptr;
+  }
+
+  return platform_context;
+}
+
+WGLPlatformContext::WGLPlatformContext() = default;
+
+WGLPlatformContext::~WGLPlatformContext() {
+  WTF_SCOPE0("WGLPlatformContext#dtor");
+
+  // Finish all context operations.
+  if (glrc_) {
+    if (MakeCurrent()) {
+      Finish();
+    }
+    ClearCurrent();
+  }
+
+  if (glrc_) {
+    xrtl_wglDeleteContext(glrc_);
+    glrc_ = nullptr;
+  }
+  if (native_display_) {
+    if (is_headless_) {
+      // Delete the DC we created for temp usage.
+      ::DeleteDC(native_display_);
+    }
+    native_display_ = nullptr;
+  }
+  if (native_window_) {
+    if (is_headless_) {
+      // Delete the temp window.
+      ::DestroyWindow(native_window_);
+    }
+    native_window_ = nullptr;
+  }
+}
+
+HWND WGLPlatformContext::CreateDummyWindow() {
+  static const wchar_t* kWindowClassName = L"XrtlWglDummyWindowClass";
+
+  // Ensure we create the window class we use for the window.
+  // This should be process-local so we only need to do it once.
+  static std::once_flag register_class_flag;
+  std::call_once(register_class_flag, []() {
+    WNDCLASSEXW wcex = {0};
+    wcex.cbSize = sizeof(WNDCLASSEXW);
+    wcex.style = CS_HREDRAW | CS_VREDRAW | CS_OWNDC;
+    wcex.lpfnWndProc = ::DefWindowProc;
+    wcex.cbClsExtra = 0;
+    wcex.cbWndExtra = 0;
+    wcex.hInstance = ::GetModuleHandle(nullptr);
+    wcex.hIcon = nullptr;
+    wcex.hIconSm = nullptr;
+    wcex.hCursor = ::LoadCursor(nullptr, IDC_ARROW);
+    wcex.hbrBackground = reinterpret_cast<HBRUSH>(COLOR_WINDOW + 1);
+    wcex.lpszMenuName = nullptr;
+    wcex.lpszClassName = kWindowClassName;
+    if (!::RegisterClassExW(&wcex)) {
+      LOG(FATAL) << "Unable to register window class";
+    }
+  });
+
+  // Setup initial window style.
+  DWORD window_style = WS_OVERLAPPEDWINDOW | WS_CLIPCHILDREN | WS_CLIPSIBLINGS;
+  DWORD window_ex_style = WS_EX_APPWINDOW | WS_EX_CONTROLPARENT;
+
+  // Create window.
+  HWND hwnd = ::CreateWindowExW(
+      window_ex_style, kWindowClassName, L"(xrtl dummy)", window_style,
+      CW_USEDEFAULT, CW_USEDEFAULT, CW_USEDEFAULT, CW_USEDEFAULT, nullptr,
+      nullptr, ::GetModuleHandle(nullptr), this);
+  if (!hwnd) {
+    DWORD last_error = ::GetLastError();
+    LOG(ERROR) << "Unable to create dummy window: "
+               << GetWglErrorName(last_error) << ": "
+               << GetWglErrorDescription(last_error);
+    return nullptr;
+  }
+
+  return hwnd;
+}
+
+bool WGLPlatformContext::Initialize(HDC native_display, HWND native_window,
+                                    ref_ptr<ES3PlatformContext> share_group) {
+  WTF_SCOPE0("WGLPlatformContext#Initialize");
+
+  // We always need a HDC/HWND to use GL. If we are offscreen we'll still create
```

Comment:
Could you add a similar comment to the ``CreateDummyWindow`` function in the header or .cc as well?